pallets/asset-index/src/lib.rs

```diff
@@ -525,13 +528,18 @@ pub mod pallet {
         /// Creates IndexAssetData if entry with given assetID does not exist.
         /// Otherwise adds the units to the existing holding
         fn add_asset(
+            metadata: AssetMetadata,
```

Comment:
This means metadata is required for every new add_asset call?
Can we extract this into a separate functions like `set_metadata`?

Reply:
> Can we extract this into a separate functions like set_metadata?

Not sure about this, for example, if we want to issue a token on ethereum, we set the metadata when we issue it

Reply:
This is different, we can add/modify the asset's metadata on the fly, we also don't need  actually require it.